implementation/src/main/java/io/smallrye/config/ConfigSourceInterceptor.java

```diff
@@ -18,6 +18,19 @@
  * order may be non deterministic.
  */
 public interface ConfigSourceInterceptor extends Serializable {
+    /**
+     * This method initializes the interceptor, with access to the current {@link ConfigSourceInterceptorContext}.
+     *
+     * Interceptors in the chain are initialized in priority order and the current
+     * {@link ConfigSourceInterceptorContext} contains the current interceptor, plus all other interceptors already
+     * initialized.
+     *
+     * @param context the current {@link ConfigSourceInterceptorContext} associated to the Config.
+     */
+    default void initialize(ConfigSourceInterceptorContext context) {
```

Comment:
I wonder if it would be better to either have a contract where interceptor constructors are invoked reflectively passing in a `ConfigSourceInterceptorContext`, or else have users implement an `ConfigSourceInterceptorFactory` interface which accepts that type and returns a `ConfigSourceInterceptor`?

This has some design benefits:
* Anything that would have been done in `initialize` can be done in a constructor and thus written to a `final` field
* Interceptors always have deterministic state
* The interceptor interface is simpler

WDYT?

Reply:
Yes, I think it would be better.

We already use a `Factory` pattern for `SmallRyeConfig`. Might as well do it here as well to keep it consistent.